File: CartService/src/main/java/com/cgi/CartService/service/CartService.java
package com.cgi.CartService.service;

import java.util.List;

import com.cgi.CartService.exceptions.CartWithTheIDAlreadyPresentException;
import com.cgi.CartService.exceptions.CartWithTheIdNotPresentException;
import com.cgi.CartService.model.Cart;



public interface CartService {
	
	
	Cart getCartById(int id) throws CartWithTheIdNotPresentException;
	Cart createNewCart(Cart cart) throws CartWithTheIDAlreadyPresentException;
	Cart deleteCart(int id) throws CartWithTheIdNotPresentException;
	Cart updateCart(Cart cart) throws CartWithTheIdNotPresentException;
}

File: jean-station-online/src/app/services/cart.service.ts

import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Cart } from '../models/Cart';
import { Product } from '../models/Product';
import { AuthenticationService } from './authentication.service';

@Injectable({
  providedIn: 'root'
})
export class CartService {
  cart(cart: any) {
    throw new Error('Method not implemented.');
  }
  

  constructor(
    private httpClient: HttpClient,
    private authentication: AuthenticationService
  ) { }



    createNewCart(cart: Cart): Observable<Cart>{
      return this.httpClient.post<Cart>(`http://localhost:9006/api/v1/cart`, cart, {
        headers: new HttpHeaders().set('Authorization', `Bearer ${this.authentication.getToken()}`)
      })
    }

    updateCart(cart: Cart): Observable<Cart>{
      return this.httpClient.put<Cart>(`http://localhost:9006/api/v1/cart/${cart.cartId}`, cart,{
        headers: new HttpHeaders().set('Authorization', `Bearer ${this.authentication.getToken()}`)
      })

    }

    getCartById(id:number): Observable<Cart> {
      return this.httpClient.get<Cart>(`http://localhost:9006/api/v1/cart/${id}`, {
        headers: new HttpHeaders().set('Authorization', `Bearer ${this.authentication.getToken()}`) 
    });

  }

    deleteCart(id: number){
      return this.httpClient.delete(`http://localhost:9006/api/v1/cart/${id}`, {
      headers: new HttpHeaders().set('Authorization', `Bearer ${this.authentication.getToken()}`)
    });
    }
  }